backend/src/main/java/com/soen341/instagram/controller/ProfileController.java

```diff
@@ -0,0 +1,128 @@
+package com.soen341.instagram.controller;
+
+// Spring Boot
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+// Project
+import com.soen341.instagram.dto.account.ProfileRequestDTO;
+import com.soen341.instagram.service.impl.ProfileService;
+
+@RestController
+public class ProfileController {
+	
+	@Autowired
+	private ProfileService profileService;
+	
+// For now we are not allowing the user to change their username
+//	@PutMapping(value = "account/profile/usernameUpdate")
+//	public void setUsername(@RequestBody final UpdateProfileRequestDTO profileRequest)
+//	{
+//		String username = profileRequest.getUsername();
+//		profileService.setUsername(username);
+//	}
+	
+	@PutMapping(value = "account/profile/biographyUpdate")
+	public void setBiography(@RequestBody final ProfileRequestDTO profileRequest)
+	{
+		String biography = profileRequest.getBiography();
+		profileService.setBiography(biography);
+	}
+	
+	@PutMapping(value = "account/profile/emailUpdate")
+	public void setEmail(@RequestBody final ProfileRequestDTO profileRequest)
+	{
+		String email = profileRequest.getEmail();
+		profileService.setEmail(email);
+	}
+	
+	@PutMapping(value = "account/profile/displayNameUpdate")
+	public void setDisplayName(@RequestBody final ProfileRequestDTO profileRequest)
+	{
+		String displayName = profileRequest.getDisplayName();
+		profileService.setDisplayName(displayName);
+	}
+	
+	@PutMapping(value = "account/profile/nameUpdate")
+	public void setName(@RequestBody final ProfileRequestDTO profileRequest)
+	{
+		String firstName = profileRequest.getFirstName();
+		String lastName = profileRequest.getLastName();
+		
+		profileService.setName(firstName, lastName);
+	}
+	
+//TODO: Implement changing profile pictures
+//	@PostMapping(value = "account/profile/newProfilePicture")
+//	public void setProfilePicture(@RequestBody final ProfilePictureDTO picture)
+//	{
+//		profileService.setProfilePicture(picture);
+//	}
+
+	@GetMapping(value = "account/profile/pictureCount")
+	public ProfileRequestDTO getNumberOfPictures()
+	{
+		ProfileRequestDTO profileRequestDTO = new ProfileRequestDTO();
+		profileRequestDTO.setNumPictures(profileService.getNumberOfPictures());
+		
+		return profileRequestDTO;
+	}
+	
+	
+	@GetMapping(value = "account/profile/followerCount")
```

Comment:
I think it would be better to simply fetch a entire Profile instead of fetching each field individually. I don't see a case where we would need to only fetch a specific field. e.g number of followers. This would limit the number of APIs we have thus reducing the numbers of calls to the server.

Reply:
It seems like a lot of information to return if we just need a small piece, but since we are returning a whole object anyways, your way is probably best. Thanks!